php-frontend/src/main/java/org/sonar/php/cfg/ControlFlowGraphBuilder.java

```diff
@@ -178,6 +186,32 @@ private static RecognitionException exception(Tree tree, Throwable cause) {
     return new RecognitionException(((PHPTree) tree).getLine(), "Failed to build CFG", cause);
   }
 
+  private PhpCfgBlock buildForEachStatement(ForEachStatementTree tree, PhpCfgBlock successor) {
+    ForwardingBlock linkToCondition = createForwardingBlock();
+
+    addBreakable(successor, linkToCondition);
+    PhpCfgBlock loopBodyBlock = buildSubFlow(tree.statements(), linkToCondition);
+    removeBreakable();
+
+    PhpCfgBranchingBlock conditionBlock = createBranchingBlock(tree, loopBodyBlock, successor);
+    conditionBlock.addElement(tree.expression());
+    linkToCondition.setSuccessor(conditionBlock);
+    return createSimpleBlock(conditionBlock);
+  }
+
+  private PhpCfgBlock buildForStatement(ForStatementTree tree, PhpCfgBlock successor) {
+    ForwardingBlock linkToCondition = createForwardingBlock();
+
+    addBreakable(successor, linkToCondition);
+    PhpCfgBlock loopBodyBlock = buildSubFlow(tree.statements(), linkToCondition);
```

Comment:
IMO we should have `update` expression somewhere here, probably it should be separate block (and be `continue` target)